### ISSUE TITLE
How to install dependencies on Ubuntu/OSX/Windows?

### ISSUE THREAD

**Issue by szabgab** [sisyphus/math-mpfr#7]
I am trying to configure GitHub Actions to run the test of this module.
It would be nice if you could include more exact instruction on how to install the external dependencies. (e.g apt-get install WHAT on debian/ubuntu)

**Comment by szabgab** [comment]
As I just found out on Ubuntu that would be installing `libgmp-dev libmpfr-dev`

**Comment by sisyphus** [comment]
I actually don't know much about installing vendor-supplied gmp and mpfr libraries.
On Ubuntu, Debian, FreeBSD and Windows (both native Windows and Cygwin) I always build the gmp and mpfr libraries directly from source.
I don't have access to systems other than those I've just specified.

I agree that it would be an improvement to do as you've suggested - and I'll happily insert that information into the beginning of the README as it comes to light. (Or is there somewhere else that it should be published ?)

I take it that on Ubuntu, the full commands are:
```
sudo apt-get install libgmp-dev
sudo apt-get install libmpfr-dev
```
I guess that the latter command would also install gmp if it wasn't already installed, but it probably doesn't hurt to provide both commands anyway.
According to what google tells me, on freebsd one would run (with suitable permissions) :
```
pkg install math/gmp
or
pkg install gmp
```
and (for the mpfr library):
```
pkg install math/mpfr
or
pkg install mpfr
```
With Cygwin, one could use the (gui) setup.exe. I don't know if there's a command line alternative.
And for native Windows, MSYS2's <code>pacman</code> would probably install suitable gmp and mpfr libraries - but that would, I think, require that MSYS2 was already installed.

Not sure about Debian ... is that going to be 'apt-get' or just plain 'apt' ? Are there some Debian systems set up to use 'yum' instead ? 

Cheers,
Rob

**Comment by szabgab** [comment]
Yeah, I think the README would be a great location. Both for vendor provided packages - where you have this information - and compile-from-source instructions. I assume Debian is the same as Ubuntu.

I sent PR #8 where I included the Ubuntu version. If you like and accept that, later we can figure out how to set up jobs with various other configurations. 

GitHub Actions currently support Ubuntu, OSX, Windows, and I think any Docker container.
